src/main/scala/wordcounter/AkkaWordCounter.scala

```diff
@@ -101,51 +116,33 @@ class Listener extends Actor {
 
 object AkkaWordCounter extends App {
 
+  import RoutingActor.StartProcessFileMsg
+  import akka.pattern.ask
   import akka.util.Timeout
+
   import scala.concurrent.duration._
-  import akka.pattern.ask
-  import akka.dispatch.ExecutionContexts._
 
   override def main(args: Array[String]) {
-
-    val files = getListOfFiles("src/main/resources/")
-
-    /**
-      * foreach takes a procedure -- a function with a result type Unit -- as the right operand.
-      * It simply applies the procedure to each List element.
-      * The result of the operation is again Unit; no list of results is assembled.
-      */
-    files.foreach(initActorSystem)
-
+    val sourceDirectoryName = if (args.length > 0) args(1) else "src/main/resources/"
+    val directory = new File(sourceDirectoryName)
+    if (directory.exists && directory.isDirectory) {
+      val system = ActorSystem()
+      implicit val ec = system.dispatcher
+      val results = directory.listFiles.map(_.getAbsolutePath).map(processFile(system, _))
+      Future.sequence(results.toIterable).onComplete { _ =>
+        system.terminate()
+      }
+    }
   }
 
-  def initActorSystem(fileName: String): Unit = {
-    //Fixing bug from original code: https://www.toptal.com/scala/concurrency-and-fault-tolerance-made-easy-an-intro-to-akka#comment-1776147740
-    implicit val executionContext = global
-    val system = ActorSystem("ActorSystem")
-    // create the result listener, which will print the result
-    val listener = system.actorOf(Props[Listener], name = "Listener")
-    //Load from /resources folder: http://stackoverflow.com/questions/27360977/how-to-read-files-from-resources-folder-in-scala
-    val stream : InputStream = getClass.getResourceAsStream("/" + fileName)
-    val actor = system.actorOf(Props(new RoutingActor(new FileReference(fileName, stream), listener)))
-    implicit val timeout = Timeout(5 seconds)
-    //When the future returns after all the work is complete
-    val futureResult = actor ? StartProcessFileMsg()
-    futureResult.map { result =>
-      //println("Number of lines processed in " + fileName + ": " + result)
-      //Terminate Actor System when result is received
-      system.terminate()
-    }
+  def processFile(system: ActorSystem, fileName: String): Future[Any] = {
+    implicit val ec = system.dispatcher
+    implicit val timeout = Timeout(30 seconds)
 
-  }
+    val listener = system.actorOf(Props[Listener], name = s"Listener:${fileName.replace('/', '_')}")
```

Comment:
When I moved to using one ActorSystem, I removed the name completely, but naming it like you did is great for debugging if something failed.